bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/PlayerActions.java

```diff
@@ -92,4 +92,20 @@ public void sendKeyRepeat(@ActionInput(name = "key") String key, @ActionInput(na
             logger.warn("Freebox Player Action service ThingHandler is null");
         }
     }
+
+    public static void sendKey(ThingActions actions, String key) {
+        ((PlayerActions) actions).sendKey(key);
```

Comment:
No need to check type?
```suggestion
        if (actions instanceof PlayerActions playerActions) {
            playerActions.sendKey(key);
        }
```

Reply:
In fact, this is the common pattern used in all bindings implementing actions.
I guess that the way the core framework is calling these static methods, the `actions` parameter will always be of the expected type.
Even if I agree with you that y-our proposal is better, is it a good idea to do it differently only in this binding ?

Reply:
Not all bindings:

https://github.com/openhab/openhab-addons/blob/858be481837ee9b45a27ffd91871728b45d1f272/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/action/EnergiDataServiceActions.java#L257-L267

I'm pretty sure I copied this pattern from another binding, but I don't remember which. Maybe deCONZ?

https://github.com/openhab/openhab-addons/blob/858be481837ee9b45a27ffd91871728b45d1f272/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/action/GroupActions.java#L142-L146

Reply:
Ok, done.